cpp/arcticdb/processing/clause_utils.cpp

```diff
@@ -180,4 +182,311 @@ template std::vector<std::vector<size_t>> structure_by_time_bucket<ResampleBound
+                const auto& field = fields.at(idx);
+                index_fields.emplace_back(field.type(), field.name());
+                // Index columns, and the first non-index column in the case of Series are always included, so remove
+                // from the column types map so they are not considered in inner/outer join
+                schema.column_types().erase(std::string(field.name()));
+            }
+            first_schema = false;
+        } else {
+            for (size_t idx = 0; idx < required_fields_count; ++idx) {
+                const auto& field = fields.at(idx);
+                auto& current_type = index_fields.at(idx).type_;
+                auto opt_common_type = has_valid_common_type(current_type, field.type());
+                if (opt_common_type.has_value()) {
+                    current_type = *opt_common_type;
+                } else {
+                    schema::raise<ErrorCode::E_DESCRIPTOR_MISMATCH>(
+                            "No common type between {} and {} when joining schemas", current_type, field.type());
+                }
+                // Index columns, and the first non-index column in the case of Series are always included, so remove
+                // from the column types map so they are not considered in inner/outer join
+                schema.column_types().erase(std::string(field.name()));
+                const auto& current_name = index_fields.at(idx).name_;
+                if (current_name != field.name()) {
+                    non_matching_name_indices.emplace(idx);
+                }
+            }
+        }
+    }
+    for (size_t idx = 0; idx < index_fields.size(); ++idx) {
+        if (non_matching_name_indices.contains(idx)) {
+            // This is the same naming scheme used in _normalization.py for unnamed multiindex levels. Ensures that any
+            // subsequent processing that checks for columns of this format will continue to work
+            stream_desc.fields().add_field(index_fields.at(idx).type(),
+                                           idx == 0 ? "index" : fmt::format("__fkidx__{}", idx));
+        } else {
+            stream_desc.add_field(index_fields.at(idx));
+        }
+    }
+    return non_matching_name_indices;
+}
+
+NormalizationMetadata generate_norm_meta(const std::vector<OutputSchema>& input_schemas,
+                                         std::unordered_set<size_t>&& non_matching_name_indices) {
+    // Ensure:
+    // All are Series or all are DataFrames
+    // All have PandasIndex OR PandasMultiIndex
+    // If PandasIndex:
+    //  - name/is_int/fake_name - if all the same maintain, otherwise "index"/false/true
+    //  - tz - if all the same maintain, otherwise empty string
+    //  - is_physically stored must all be the same
+    //  - RangeIndex
+    //    - start==0/step==1 - maintain
+    //    - All steps the same, use start from first schema and maintain step
+    //    - Otherwise, log warning, set start==0/step==1
```

Comment:
How do we fill in the gaps in this case? If symbol 1 has start=0 stop=10 step=1 and symbol 2 has start=10, stop=20 step =2, what values do we return for 11, 13 etc.?